.github/workflows/ci.yml

```diff
@@ -9,64 +9,105 @@ on:
+
+      - name: CLANG/Release/cmake
+        run: export CC=/usr/bin/clang && cmake -DCMAKE_BUILD_TYPE=Release ..
+        working-directory: /home/runner/work/pgexporter/pgexporter/build/
+
+      - name: CLANG/Release/make
+        run: make
+        working-directory: /home/runner/work/pgexporter/pgexporter/build/
+
+  build-macos:
+    runs-on: macos-latest
     steps:
-    - uses: actions/checkout@v3
-    - name: Update system
-      run: sudo apt update -y
-    - name: Install libev
-      run: sudo apt install -y libev4 libev-dev
-    - name: Install systemd
-      run: sudo apt install -y libsystemd-dev
-    - name: Install rst2man
-      run: sudo apt install -y python3-docutils
-    - name: Install zstd
-      run: sudo apt install -y libzstd-dev
-    - name: Install lz4
-      run: sudo apt install -y liblz4-dev
-    - name: Install bzip2
-      run: sudo apt install -y libbz2-dev
-    - name: Install graphviz
-      run: sudo apt install graphviz
-    - name: Install doxygen
-      run: sudo apt install doxygen
-    - name: Install libyaml
-      run: sudo apt install -y libyaml-dev
-    - name: Install clang
-      run: sudo apt install -y clang
-    - name: GCC/mkdir
-      run: mkdir build
-      working-directory: /home/runner/work/pgexporter/pgexporter/
-    - name: GCC/cmake
-      run: export CC=/usr/bin/gcc && cmake -DCMAKE_BUILD_TYPE=Debug ..
-      working-directory: /home/runner/work/pgexporter/pgexporter/build/
-    - name: GCC/make
-      run: make
-      working-directory: /home/runner/work/pgexporter/pgexporter/build/
-    - name: rm -Rf
-      run: rm -Rf build/
-      working-directory: /home/runner/work/pgexporter/pgexporter/
-    - name: CLANG/mkdir
-      run: mkdir build
-      working-directory: /home/runner/work/pgexporter/pgexporter/
-    - name: CLANG/cmake
-      run: export CC=/usr/bin/clang && cmake -DCMAKE_BUILD_TYPE=Debug ..
-      working-directory: /home/runner/work/pgexporter/pgexporter/build/
-    - name: CLANG/make
-      run: make
-      working-directory: /home/runner/work/pgexporter/pgexporter/build/
-    - name: rm -Rf
-      run: rm -Rf build/
-      working-directory: /home/runner/work/pgexporter/pgexporter/
-    - name: CLANG/Release/mkdir
-      run: mkdir build
-      working-directory: /home/runner/work/pgexporter/pgexporter/
-    - name: CLANG/Release/cmake
-      run: export CC=/usr/bin/clang && cmake -DCMAKE_BUILD_TYPE=Release ..
-      working-directory: /home/runner/work/pgexporter/pgexporter/build/
-    - name: CLANG/Release/make
-      run: make
-      working-directory: /home/runner/work/pgexporter/pgexporter/build/
+      - uses: actions/checkout@v3
+
+      - name: Install dependencies
+        run: |
+          brew update
+          brew install libev zstd lz4 bzip2 graphviz doxygen libyaml cmake
+
+      - name: Clang/Debug/mkdir
+        run: mkdir build
+        working-directory: /home/runner/work/pgexporter/pgexporter/
+
+      - name: Clang/Debug/cmake
```

Comment:
I think [this](https://apple.stackexchange.com/a/40737) should serve the purpose better?